src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -114,7 +114,8 @@ internal partial class Binder
             ConstantValue constantValue = this.FoldConstantConversion(syntax, source, conversion, destination, diagnostics);
             if (conversion.Kind == ConversionKind.DefaultOrNullLiteral && source.Kind == BoundKind.DefaultExpression)
             {
-                source = ((BoundDefaultExpression)source).Update(constantValue, destination);
+                var defaultExpression = (BoundDefaultExpression)source;
+                source = defaultExpression.Update(defaultExpression.IsNullable, constantValue, destination);
```

Comment:
Should `Update` just maintain the `IsNullable` state by default? (ie only update the members that need to be updated)